kernel/sem.c

```diff
@@ -60,6 +61,7 @@ void _impl_k_sem_init(struct k_sem *sem, unsigned int initial_count,
 	__ASSERT(limit != 0, "limit cannot be zero");
 	__ASSERT(initial_count <= limit, "count cannot be greater than limit");
 
+	sys_trace_void(SYS_TRACE_ID_SEMA_INIT);
```

Comment:
Could we attach some information to the trace packet in order to identify given semaphore in trace log?

Reply:
yes, the POC started with something simple to get the general concept in place. We can add the semaphore info as a payload.